labs/src/main/scala/org/scalalabs/basic/lab02/ListManipulationExercise02.scala

```diff
@@ -11,7 +11,11 @@ object ListManipulationExercise02 {
    * As usual, various ways exist: pattern matching, folding, ...
    */
   def maxElementInList(l: List[Int]): Int = {
-    error("fix me")
+    l match{
+      case List(x:Int) => x
+      case x :: y :: xs => maxElementInList((if (x>y) x else y)::xs)
+      case nil => throw new IllegalArgumentException("empty list")
```

Comment:
I don't think I solved this one with pattern matching, but I'd believe that incomplete pattern matching causes an error.